extra_data/file_access.py

```diff
@@ -122,8 +123,13 @@ class FileAccess(metaclass=MetaFileAccess):
     """
     _file = None
     _format_version = None
+    _path_traits = None
+    _filename_traits = None
     metadata_fstat = None
 
+    exdf_path_pattern = re.compile(r'\/(gpfs\/exfel\/d|pnfs\/xfel.eu\/exfel\/archive\/XFEL)\/([a-z]+)\/(\w+)\/(\d{6})\/p\d{6}\/r\d{4}')
```

Comment:
This won't work on ONC, where the paths really are under `/gpfs/exfel/exp`, with the components in a different order from this.

I'm OK with this - clearly the path based info can't always work, because we allow opening files & folders from anywhere. So I just want to ensure we're making a conscious choice, not expecting something to work which won't.